core/src/block_sync.rs

```diff
@@ -25,66 +26,100 @@ pub struct BlockSynchronizer {
     peer_id: PeerId,
     gossip_period: Duration,
     block_batch_size: u32,
-    broker: Broker,
-    actor_channel_capacity: u32,
-}
-
-#[async_trait::async_trait]
-impl Actor for BlockSynchronizer {
-    fn actor_channel_capacity(&self) -> u32 {
-        self.actor_channel_capacity
-    }
+    p2p: Arc<P2PSystem>,
+    /// Sender channel
+    pub message_sender: Mutex<mpsc::SyncSender<message::VersionedMessage>>,
```

Comment:
Yet another pair of `Mutex`. Why not just have a single `Mutex` with a tuple of two channels?

Reply:
Better yet, consider alternatives which don't need to be wrapped in a `Mutex`.